search-sentiments-map/test/country-trends-test.js

```diff
@@ -0,0 +1,189 @@
+      },
+      {
+        country: 'AR',
+        trends: [{
+          topic: 'Dogs',
+        }],
+      }];
+      mockData = [{
+        timestamp: currentTime - THREE_HALVES_TIME_RANGE *
+            CURRENT_DATA_TIME_RANGE_12_HOURS_MS,
+        trendsByCountry: trendsByCountryThreeHalvesTimeRange,
+        globalTrends: [],
+      },
+      {
+        timestamp: currentTime - THREE_TIME_RANGE *
+            CURRENT_DATA_TIME_RANGE_12_HOURS_MS,
+        trendsByCountry: trendsByCountryThreeTimeRange,
+        globalTrends: [],
+      }];
+
+      mockThreeHalvesTimeRangeUsResult = [
+        {topic: 'Live'}, 
+        {topic: 'Laugh'}
+      ];
+      mockThreeTimeRangeUsResult = [
+        {topic: 'Donald Trump'}, 
+        {topic: 'Coronavirus'}
+      ];
+
+      mockThreeHalvesTimeRangeUaResult = [
+        {topic: 'Love'}, 
+        {topic: 'Black Lives Matter'}
+      ];
+      mockThreeTimeRangeUaResult = [
+        {topic: 'Hello World'}, 
+        {topic: 'Coronavirus'}
+      ];
+
+      mockEmptyResult = [];
+
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        return [mockData];
+      });      
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should retrieve the most recent array of US and UA country trends from the datastore beyond the time it takes to retrieve search results', 
+        async function() {
+      let resultsUs = await countryTrends.retrieveCountryTrends('US', 0);
+      let resultsUa = await countryTrends.retrieveCountryTrends('UA', 0);
+      assert.deepEqual(resultsUs, mockThreeHalvesTimeRangeUsResult);
+      assert.deepEqual(resultsUa, mockThreeHalvesTimeRangeUaResult);
+    });
+
+    it('should retrieve the most recent array of US and UA country trends from the datastore beyond the given time range', 
+        async function() {
+      let resultsUs = await countryTrends.retrieveCountryTrends('US', 2);
+      let resultsUa = await countryTrends.retrieveCountryTrends('UA', 2);
+      assert.deepEqual(resultsUs, mockThreeTimeRangeUsResult);
+      assert.deepEqual(resultsUa, mockThreeTimeRangeUaResult);
+    });
+
+    it('should get an empty array because there is no trends data available for AT', 
+        async function() {
+      let resultsAt = await countryTrends.retrieveCountryTrends('AT', 0);
+      assert.deepEqual(resultsAt, mockEmptyResult);
+    });
+  });
+
+  describe('retrieveCountryTrendsNoDataInDatastore', function() {
+    let mockEmpty;
+    beforeEach(() => {
+      mockEmpty = [];
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        return [mockEmpty];
+      });      
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should get an empty array', async function() {
+      let result = await countryTrends.retrieveCountryTrends('US', 0);
+      assert.deepEqual(result, mockEmpty);
+    });
+  });
+});
```

Comment:
If you are testing the same function, as the describe function above, this should actually be in the same describe block, but a different it block. And the the description of the it block should be "should return an empty array when there are no trends in the datastore"